resources/js/components/sponsors-dashboard/components/Resources/SponsorPortal.tsx

```diff
@@ -16,7 +16,7 @@ interface ISponsorPortalProps extends RouteComponentProps {
 }
 
 interface ISponsorPortalState {
-    detail_id: string,
+    detail_id: number,
```

Comment:
Ah, so yeah - this was causing a bug. The id needs to be a string as this is what the backend gives us. It is a number in a string type to be more precise.

There was a bug with this which every time it created new entries into the sponsors database when you save and it was because it could not recognise the id.

Reply:
Yeah exactly.

Reply:
Wait, so does it work for you when it is a `number`, as for me it works when it is a `string`?

Reply:
Wait really? Yeah it definitely works on my end. Though I suppose this isn't a strictly typed so it wouldn't matter...it's more the `setState` that does it. It is similarly a `number` in other parts of the codebase, such s `ISingleItemFormState`

`nterface ISingleItemFormState {
    detail_id: number,
    complete: string,
    fields: SingleItemFormFields,
    uploadFormShowing: boolean,
    showDestructiveForm: JSX.Element | undefined,
    isLoading: boolean,
}`

Reply:
We found a way to resolve it